cilium/cmd/monitor.go

```diff
@@ -175,12 +175,35 @@ func getMonitorParser(conn net.Conn, version listener.Version) (parser eventPars
 	}
 }
 
+func endpointExists(endpoint format.Uint16Flags) bool {
+	eps, err := client.EndpointList()
+	if err != nil {
+		Fatalf("cannot get endpoint list: %s\n", err)
+	}
+	for _, ep := range eps {
+		if endpoint.Has(uint16(ep.ID)) {
+			return true
+		}
+	}
+	return false
+}
+
 func runMonitor(args []string) {
 	if len(args) > 0 {
 		fmt.Println("Error: arguments not recognized")
 		os.Exit(1)
 	}
 
+	if len(printer.FromSource) > 0 && !endpointExists(printer.FromSource) {
+		Fatalf("Endpoint %d doesn't exist\n", printer.FromSource[0])
```

Comment:
If this is changed per my suggestion above, then the `[0]` probably isn't strictly correct any more. (Same for the other instances below)